include/maths/CTimeSeriesDecomposition.h

```diff
@@ -166,6 +166,9 @@ class MATHS_EXPORT CTimeSeriesDecomposition : public CTimeSeriesDecompositionInt
     virtual maths_t::TDoubleDoublePr
     scale(core_t::TTime time, double variance, double confidence, bool smooth = true) const;
 
+    //! Get the values in a recent time window if they are available.
+    virtual TTimeDoublePrVec windowValues(core_t::TTime time, bool forced = false) const;
```

Comment:
Could you document what forcing means here?

Reply:
On reflection I think a slight refactor makes this easier to understand. I factored out the logic to test to see if the components might be added so this now always returns the decomposition window values. See [this](https://github.com/elastic/ml-cpp/pull/218/commits/01ff26346f4464f8428e8515ac9da7515f18b7ca) commit.